misc/python/materialize/mzcompose/services/sql_logic_test.py

```diff
@@ -14,20 +14,20 @@
 from materialize.mzcompose.service import (
     Service,
 )
-from materialize.mzcompose.services.postgres import METADATA_STORE
 
 
 class SqlLogicTest(Service):
     def __init__(
         self,
         name: str = "sqllogictest",
         mzbuild: str = "sqllogictest",
-        environment: list[str] = [
-            "MZ_SOFT_ASSERTIONS=1",
-        ],
+        environment: list[str] | None = None,
         volumes: list[str] = ["../..:/workdir"],
-        depends_on: list[str] = [METADATA_STORE],
     ) -> None:
+        if not environment:
+            environment = [
+                "MZ_SOFT_ASSERTIONS=1",
+            ]
```

Comment:
nit: I might be missing something but is it worth just leaving the environment default as is we are going to default it to that value for the new default value? totally possibly I'm just not getting how None works.

Reply:
This was actually a footgun/bug. In Python having a list as a default parameter is very dangerous, because if you modify it inside the function, you also modify the default for future calls to it.